### PR TITLE
Add uutils coreutils

### PR DESCRIPTION
**Pull request by sshockwave**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ x License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sshockwave** [comment]
Ready for review!

@conda-forge-admin, please ping conda-forge/staged-recipes

File: recipes/uutils/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{% set name = "uutils" %}
```
Comment (carterbox) [inline recipes/uutils/meta.yaml:1]: Shouldn't the name of this package be `uutils-coreutils`? We already have a `GNU-coreutils` package. This is a replacement for that package, right? If the have the same API/named outputs, then you also need to make these packages mutually exclusive using `run_constrained` so they don't squash each other's artifacts.

https://github.com/conda-forge/coreutils-feedstock
https://conda-forge.org/feedstock-outputs/

File: recipes/uutils/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    - make
+    - cargo-bundle-licenses
+
+test:
+  commands:
+    - coreutils --help
+    - test -f "${PREFIX}"/bin/coreutils  # [not win]
+    - test -f "${PREFIX}"/bin/cat  # [not win]
+    - if not exist %LIBRARY_BIN%\coreutils.exe exit 1  # [win]
+    - if not exist %LIBRARY_BIN%\cat.exe exit 1  # [win]
```
Comment (carterbox) [inline recipes/uutils/meta.yaml:34]:
```suggestion
    - coreutils --help
    - cat --help
```
This should be sufficient.

File: recipes/uutils/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  sha256: 1bdc2838e34b6ca8275cb8e3bac5074ca59158ea077e3195fb91ec0cc6b594f6
+  patches:
+    - 0001-FIX_BUILDDIR_IN_MAKE.patch  # [not win]
+    # Fixes long command and quotes problems
+    - 0002-SHORTEN_SH_COMMANDS.patch
+
+build:
+  number: 0
+  ignore_run_exports:
+    - libcxx-ng
```
Comment (carterbox) [inline recipes/uutils/meta.yaml:19]:
```suggestion
  ignore_run_exports_from:
    - {{ compiler('cxx') }}
```
The cxx standard library is also not used.

**Review by carterbox — CHANGES_REQUESTED** [review]

**Comment by carterbox** [comment]
Thanks for gettting the licensing right!

File: recipes/uutils/meta.yaml (hunk 1 above)
Comment (sshockwave) [reply recipes/uutils/meta.yaml:1]: Adding the `-coreutils` suffix would be a nice hint of clobbering artifacts. Thanks for the careful review!

**Comment by sshockwave** [comment]
@carterbox I might need some help here. The package needs cxx compiler to compile. It also depends on `libgcc`, but not `libcxx`. Is it ok to leave `libcxx` as an unused dependency?

**Comment by carterbox** [comment]
No. Then add more packages to `ignore_run_exports` appropriately to avoid dependencies on the unused libs.

**Comment by sshockwave** [comment]
@conda-forge/help-c-cpp ready for review!

File: recipes/uutils/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    - libstdcxx-ng  # [linux]
+  ignore_run_exports_from:
+    - {{ compiler('cxx') }}  # [not linux]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('rust') }}
+    - make
+    - cargo-bundle-licenses
```
Comment (carterbox) [inline recipes/uutils/meta.yaml:28]:
```suggestion
requirements:
  build:
    - {{ compiler('cxx') }}
    - {{ compiler('rust') }}
    - make
    - cargo-bundle-licenses
  run_constrained:
    - coreutils ==999999999
    - gnu-coreutils ==999999999
```
I assume that your package clobbers files from these packages, so they should never be installed into the same environment.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/uutils/meta.yaml (hunk 4 above)
Comment (sshockwave) [reply recipes/uutils/meta.yaml:28]: Thank you! I didn't realize that `run_constrained`needs to be added in this `meta.yaml`.

**Pull request merged**